pkg/volume/cinder/attacher.go

```diff
@@ -318,3 +332,23 @@ func (attacher *cinderDiskAttacher) nodeInstanceID(nodeName types.NodeName) (str
 	}
 	return instanceID, nil
 }
+
+//Get the node instanceID from node spec
+func getNodeInstanceID(host volume.VolumeHost, nodeName types.NodeName) (string, error) {
+	kclient := host.GetKubeClient()
+	if kclient == nil {
+		return "", fmt.Errorf("Cannot get kube client")
+	}
+
+	node, err := kclient.Core().Nodes().Get(string(nodeName), metav1.GetOptions{})
```

Comment:
The idea was to avoid a lookup :(
The full node object almost certainly *already exists* further up the callstack somewhere, and it already includes the `.Spec.ProviderID` property that we need.  So we should modify the *callers* of `Attach`/`Detach`/etc to pass that full node object (or just instance ID?), rather than a rather useless node name into those functions.